ecommerce-website/src/containers/Products/Products.js

```diff
@@ -0,0 +1,113 @@
+import React,{Component} from 'react';
+import Product from '../../components/Product/Product';
+import styles from './Products.module.css';
+import axios from 'axios';
+import Loader from '../../components/ui/loader/Loder';
+import SearchSort from '../../components/ui/SearchSort/SearchSort';
+
+class Products extends Component{
+    state={
+        products:null,
+        error:false,
+        loading:true,
+        productsbycategory:null,
+        searchProduct:null
+    }
+    componentDidMount(){ 
+        axios.get( 'https://fakestoreapi.com/products' )
+            .then( response => {
+                this.setState({products:response.data,loading:false})
+            } )
+            .catch( error => {
+                this.setState({error:true,loading:false})
+            } );
+    }
+    clickHandler=(id)=>{
+        this.props.history.push(`/products/${id}`)
+    }
+    sortHandler = (e)=>{
+        let arrOfSortedProduct = [];
+        let products=[...this.state.products];
+        for(let i=0;i<products.length;i++){
+            if(products[i].category === e.target.value){
+                arrOfSortedProduct.push(products[i]);
+            } 
+            if(e.target.value === 'All'){
+                arrOfSortedProduct.push(products[i]);
+            }
+            if(e.target.value === 'under 50$'){
+                if(products[i].price <= 50){
+                    arrOfSortedProduct.push(products[i]);
+                }
+            }
+            if(e.target.value === '100$-150$'){
+                if(products[i].price > 100 && products[i].price <= 150){
+                    arrOfSortedProduct.push(products[i]);
+                }
+            }
+            if(e.target.value === '50$-100$'){
+                if(products[i].price > 50 && products[i].price <= 100){
+                    arrOfSortedProduct.push(products[i]);
+                }
+            }
+        }
+        this.setState({productsbycategory:arrOfSortedProduct})
+    }
+    searchHandler = (event) =>{
+        let products=[...this.state.products];
+        let searchedProductInfo;
+        products.forEach(data=>{
```

Comment:
forEach would run for length of the array even if condition is satisfied. Can we use filter()?